src/main/java/net/openhft/chronicle/jlbh/PercentileSummary.java

```diff
@@ -0,0 +1,102 @@
+package net.openhft.chronicle.jlbh;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * Calculates values for the run summary.
+ * <p>
+ * Handle runs with differing number of samples (and hence, percentiles) correctly
+ */
+public class PercentileSummary {
+
+    private final boolean skipFirst;
+    private final List<double[]> percentileRuns;
+    private final double[] percentiles;
+
+    public PercentileSummary(boolean skipFirst,
```

Comment:
nice cleanup

Reply:
Would it be possible to use this to get rid of TeamCityHelper.shortenArray?

Reply:
I will look into that